agenta-cli/agenta/sdk/tracing/inline.py

```diff
@@ -0,0 +1,1305 @@
+############################
+### services.shared.dtos ###
+### -------------------- ###
+
+from typing import Optional
+
+from pydantic import BaseModel
+from uuid import UUID
+from datetime import datetime
+from enum import Enum
+from collections import OrderedDict
+
+
+NOF_CHARS = 8
+
+
+def _p_id(id):
+    return repr(str(id)[:NOF_CHARS])
+
+
+def _p_osa(o):
```

Comment:
maybe add some docstring to this one and the next

Reply:
gotcha